registrar.go

```diff
@@ -53,11 +54,15 @@ func (e *RegisteredEntity) SetVersion(v int32) {
 
 // EntityInfo is a helper for accessing the registered entity's EntityInfo
 // instance which is required by clients to call connector methods.
-func (e *RegisteredEntity) EntityInfo() *EntityInfo {
+func (e *RegisteredEntity) EntityInfo(dynTTL *time.Duration) *EntityInfo {
```

Comment:
Keep the original EntityInfo method interface. Change the TTL after you get the entityinfo instance since anyway the TTL is a public field.